ui/common/src/events.ts

```diff
@@ -0,0 +1,32 @@
+import type { Hooks } from 'snabbdom';
+
+export function bindWhileAlive(
+  el: HTMLElement,
+  event: string,
+  onInsert: (v: HTMLElement) => (e: Event) => void
+): Hooks {
+  let unbind: (() => void) | undefined;
+
+  return {
+    insert: vnode => {
+      unbind?.();
+      const callback = onInsert(vnode.elm as HTMLElement);
+      el.addEventListener(event, callback);
+      unbind = () => el.removeEventListener(event, callback);
```

Comment:
is it necessary to remove the event listener from a DOM element that is about to be destroyed? I'm pretty sure the browser does that on its own